Modules/selectmodule.c

```diff
@@ -221,7 +221,7 @@ select_select(PyObject *self, PyObject *args)
             return NULL;
         }
 
-        if (_PyTime_AsTimeval(timeout, &tv, _PyTime_ROUND_CEILING) == -1)
+        if (_PyTime_AsTimeval(timeout, &tv, _PyTime_ROUND_UP) == -1)
```

Comment:
`timeout` above is rounded with `_PyTime_ROUND_CEILING`. Negative values between -1e-9 and 0 already are rounded to 0.

Reply:
Done in 5c96cd6

Reply:
Shouldn't `_PyTime_ROUND_UP` be used instead in `_PyTime_FromSecondsObject()`?

Reply:
Yes! Corrected in 3f1f55353d. I am sorry if I miss a few places. A combination of ROUND_CEILING and ROUND_UP was causing a rounding error in test.eintrdata.eintr_tester.TimeEINTRTest and I am a bit paranoid changing these. :disappointed: